frontend/src/components/file-explorer/TreeNode.test.tsx

```diff
@@ -1,148 +1,121 @@
 import React from "react";
-import { act, render } from "@testing-library/react";
+import { act } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { renderWithProviders } from "test-utils";
 import TreeNode from "./TreeNode";
-import { WorkspaceFile } from "#/services/fileService";
-
-const onFileClick = vi.fn();
-
-const NODE: WorkspaceFile = {
-  name: "folder",
-  children: [
-    { name: "file.ts" },
-    { name: "folder2", children: [{ name: "file2.ts" }] },
-  ],
-};
+import { selectFile, listFiles } from "#/services/fileService";
+
+vi.mock("../../services/fileService", async () => ({
+  listFiles: vi.fn(async (path: string = "/") => {
+    if (path === "/") {
+      return Promise.resolve(["folder1/", "file1.ts"]);
+    }
+    if (path === "/folder1/") {
+      return Promise.resolve(["file2.ts"]);
+    }
+    return Promise.resolve([]);
+  }),
```

Comment:
```suggestion
  listFiles: vi.fn().mockResolvedValue(["file2.ts"]),
```

TIP: If most of your tests expect `listFiles` to return `["file2.ts"]`, you can directly set it here. If there are certain cases where you want it to resolve to a different value, in that test, you can do something like this:

```ts
it("should resolve to a different value", () => {
  (listFiles as Mock).mockResolvedValueOnce(["foo.ts", "bar.ts"]);
  /* ... */
});
```

Also, pay extra attention that I am using `.mockResolvedValue` rather than `.mockImplementation` or `.mockReturnValue`. This already resolves a promise. See more mock functions [here](https://vitest.dev/api/mock)

Reply:
This is probably better as-is--the idea is that we specify the full file structure here using if/else, in once place